### PR TITLE
Add (efficiency and load regulation) and (line regulation) measurements.

### PR DESCRIPTION
**Pull request by PJarali-NI**
# Add two measurements: -

### Power on the DUT
image (44)

### Perform Measurement
image (45)
image (51)
image (52)

### Power off the DUT
image (46)

### Power on the DUT
image (47)

### Perform Measurement
image (48)
image (49)

### Power off the DUT
image (50)

### CODE REVIEW

File: source/line regulation/LineRegulation_PMIC.measui
Comment (ngoliNI) [inline source/line regulation/LineRegulation_PMIC.measui:1]: Load voltage deviation (%) is a percentage value there shouldn't be any prefix/units. can you change the display format of that indicator.

File: source/line regulation/configure_dc_power.py [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+# Sweep type indicator ENUM
+class SweepType(Enum):
+    Linear = 0
+    Logarithmic = 1
+    pass
+
+
+# function to take voltage and current levels and return status string
+def format_power_on_result(voltage_level: float, current_level: float) -> str:
```
Comment (ngoliNI) [inline source/line regulation/configure_dc_power.py:26]: Can you limit both Voltage Level and Current Level to 3 decimal points

**Review by ngoliNI — CHANGES_REQUESTED** [review]

File: source/line regulation/configure_dc_power.py (hunk 1 above)
Comment (PJarali-NI) [reply source/line regulation/configure_dc_power.py:26]: Changed them to display only 3 digits after decimal point.

File: source/line regulation/LineRegulation_PMIC.measui
Comment (PJarali-NI) [reply source/line regulation/LineRegulation_PMIC.measui:1]: Removed % symbol from indicator.

**Review by Upatel-NI — CHANGES_REQUESTED** [review]

File: source/efficiency and load regulation/configure_dc_power.py [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+
+        session.channels[channel_name].current_limit = current_limit
+        session.channels[channel_name].voltage_level = voltage_level
+
+        session.channels[channel_name].commit()
+        session.channels[channel_name].measure_when = MeasureWhen.ON_DEMAND
+        session.channels[channel_name].initiate()
+
+        session.channels[channel_name].wait_for_event(event_id=Event.SOURCE_COMPLETE, timeout=5)
+        voltage = session.channels[channel_name].measure(measurement_type=MeasurementTypes.VOLTAGE)
```
[the remark this answers is not in the document]
Comment (PJarali-NI) [reply source/efficiency and load regulation/configure_dc_power.py:112]: This is done on UI side

File: source/efficiency and load regulation/configure_dc_power.py [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+        current = session.channels[channel_name].measure(measurement_type=MeasurementTypes.CURRENT)
```
[the remark this answers is not in the document]
Comment (PJarali-NI) [reply source/efficiency and load regulation/configure_dc_power.py:113]: This is done on UI side

File: source/efficiency and load regulation/EfficiencyAndLoadRegulation_PMIC.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+script_or_exe = sys.executable if getattr(sys, "frozen", False) else __file__
+service_directory = pathlib.Path(script_or_exe).resolve().parent
+measurement_service = nims.MeasurementService(
+    service_config_path=service_directory / "EfficiencyAndLoadRegulation_PMIC.serviceconfig",
+    version="1.0.0.0",
+    ui_file_paths=[service_directory / "EfficiencyAndLoadRegulation_PMIC.vi"],
+)
+
+
```
[the remark this answers is not in the document]
Comment (PJarali-NI) [reply source/efficiency and load regulation/EfficiencyAndLoadRegulation_PMIC.py:18]: Added config comments

File: source/efficiency and load regulation/configure_dc_power.py [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+            session.channels[channel_name].current_limit = get_current_limit(i, current_limit, power_limit)
+
+    session.channels[channel_name].measure_when = MeasureWhen.AUTOMATICALLY_AFTER_SOURCE_COMPLETE
+    session.channels[channel_name].aperture_time = aperture_time
+
+    session.channels[channel_name].commit()
+    return
+
+
+# function to configure source for perform measurement
```
[the remark this answers is not in the document]
Comment (PJarali-NI) [reply source/efficiency and load regulation/configure_dc_power.py:262]: changed the comment appropriately

File: source/efficiency and load regulation/configure_dc_power.py [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+def configure_load(
+        session: Session,
+        channel_name: str,
+        current_levels: list[float],
+        voltage_limit_range: float,
+        aperture_time: float,
+        source_terminal_name: str,
+        measure_terminal_name: str
+) -> None:
+    # configure the source session
```
[the remark this answers is not in the document]
Comment (PJarali-NI) [reply source/efficiency and load regulation/configure_dc_power.py:272]: changed the comment appropriately

File: source/efficiency and load regulation/configure_dc_power.py [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+            if channel_name[j].isnumeric():
+                while j < len(channel_name) and channel_name[j].isnumeric():
+                    res += channel_name[j]
+                    j += 1
+                break
+            j += 1
+
+        return f'/{resource_descriptor}/Engine{res}/{local_terminal_name}'
+
+    elif model == 'NI PXI-4132':
```
[the remark this answers is not in the document]
Comment (PJarali-NI) [reply source/efficiency and load regulation/configure_dc_power.py]: function has been changed

File: source/line regulation/LineRegulation_PMIC.py [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+
+script_or_exe = sys.executable if getattr(sys, "frozen", False) else __file__
+service_directory = pathlib.Path(script_or_exe).resolve().parent
+measurement_service = nims.MeasurementService(
+    service_config_path=service_directory / "LineRegulation_PMIC.serviceconfig",
+    version="1.0.0.0",
+    ui_file_paths=[service_directory / "LineRegulation_PMIC.measui"],
+)
+
+
```
[the remark this answers is not in the document]
Comment (PJarali-NI) [reply source/line regulation/LineRegulation_PMIC.py:18]: added comments like OVA

File: source/efficiency and load regulation/configure_dc_power.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (PJarali-NI) [reply source/efficiency and load regulation/configure_dc_power.py:112]: image

File: source/efficiency and load regulation/configure_dc_power.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (PJarali-NI) [reply source/efficiency and load regulation/configure_dc_power.py:113]: image

**Review by Upatel-NI — APPROVED** [review]

**Pull request merged**